lib/components/base-components/NormalComponent/NormalComponent_doInitialPcbFootprintStringRender.ts

```diff
@@ -140,19 +140,28 @@ export function NormalComponent_doInitialPcbFootprintStringRender(
 
     // Find resolver: library can be a function or an object of resolvers
     let resolverFn:
-      | ((path: string) => Promise<FootprintLibraryResult | any[]>)
+      | ((
+          path: string,
+          options?: { resolvedPcbStyle?: any },
+        ) => Promise<FootprintLibraryResult | any[]>)
       | undefined
     if (typeof libMap === "function") {
       resolverFn = libMap as (
         path: string,
+        options?: { resolvedPcbStyle?: any },
       ) => Promise<FootprintLibraryResult | any[]>
     }
 
     if (!resolverFn) return
 
+    // Get the resolved pcbStyle from the component
+    const resolvedPcbStyle = component.getInheritedProperty("pcbStyle")
```

Comment:
you actually can't use `getInheritedProperty` for `pcbStyle`, you have to merge each parent i.e. the resolved style is like...

```tsx
{
  ...child.parent.parent.pcbStyle,
  ...child.parent.pcbStyle,
  ...child.pcbStyle
}
```

You should introduce a new function called `getInheritedMergedProperty("properyName")`